addons/website_event_exhibitor/controllers/exhibitor.py

```diff
@@ -77,16 +78,22 @@ def _event_exhibitors_get_values(self, event, **searches):
         sponsor_types = sponsors_all.mapped('sponsor_type_id')
         sponsor_countries = sponsors_all.mapped('partner_id.country_id').sorted('name')
         # organize sponsors into categories to help display
-        sponsor_categories = dict()
+        sponsor_categories_dict = defaultdict(request.env['event.sponsor'].sudo)
+        sponsor_categories = []
+        is_event_user = request.env.user.has_group('event.group_event_user')
         for sponsor in sponsors:
-            if not sponsor_categories.get(sponsor.sponsor_type_id):
-                sponsor_categories[sponsor.sponsor_type_id] = request.env['event.sponsor'].sudo()
-            sponsor_categories[sponsor.sponsor_type_id] |= sponsor
-        sponsor_categories = [
-            dict({
+            sponsor_categories_dict[sponsor.sponsor_type_id] |= sponsor
+
+        for sponsor_category, sponsors in sponsor_categories_dict.items():
+            # To display random published sponsors first and random unpublished sponsors last
+            if is_event_user:
+                published_sponsors = sponsors.filtered(lambda s: s.is_published)
```

Comment:
Could probably be done using a sorted

Reply:
Related to above point, it's used for showing random sponsors but the code was lost.

Reply:
:wave: is there any reason here to use `is_published`?
`is_published` should never be used but `website_published` instead
even if in this case it is not the multi mixin, so it is just a related, `website_published` is better for consistency in the code
(except on the `order` key from the `search()` where it is not possible to use it)

Reply:
:+1: Use website_published

Reply:
Done